src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp_pkey_dsa.c

```diff
@@ -2,13 +2,147 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 #include "pal_evp_pkey_dsa.h"
+#include "pal_utilities.h"
+#include <assert.h>
 
-DSA* CryptoNative_EvpPkeyGetDsa(EVP_PKEY* pkey)
+EVP_PKEY* CryptoNative_EvpPKeyCreateDsa(DSA* currentKey)
 {
-    return EVP_PKEY_get1_DSA(pkey);
+    assert(currentKey != NULL);
+
+    EVP_PKEY* pkey = EVP_PKEY_new();
+
+    if (pkey == NULL)
+    {
+        return NULL;
+    }
+
+    if (!EVP_PKEY_set1_DSA(pkey, currentKey))
+    {
+        EVP_PKEY_free(pkey);
+        return NULL;
+    }
+
+    return pkey;
 }
 
-int32_t CryptoNative_EvpPkeySetDsa(EVP_PKEY* pkey, DSA* dsa)
+EVP_PKEY* CryptoNative_DsaGenerateKey(int32_t keySize)
 {
-    return EVP_PKEY_set1_DSA(pkey, dsa);
+    EVP_PKEY_CTX* paramCtx = EVP_PKEY_CTX_new_id(EVP_PKEY_DSA, NULL);
+
+    if (paramCtx == NULL)
+    {
+        return NULL;
+    }
+
+    EVP_PKEY* paramKey = NULL;
+    EVP_PKEY* pkey = NULL;
+    EVP_PKEY* ret = NULL;
+    EVP_PKEY_CTX* keyCtx = NULL;
+
+    // FIPS 186-4 4.2
+    // int qbits = keySize > 1024 ? 256 : 160;
+
+    if (EVP_PKEY_paramgen_init(paramCtx) == 1 && EVP_PKEY_CTX_set_dsa_paramgen_bits(paramCtx, keySize) == 1 &&
+        // EVP_PKEY_CTX_set_dsa_paramgen_q_bits(paramCtx, qbits) == 1 &&
```

Comment:
Looks like I knew it was potentially a problem, but ultimately decided it wasn't needed.  Well.  That saves some work ;)